### PR TITLE
[#3457]Create new resource view if resource format changed

### PR DESCRIPTION
**Pull request by Engerrs**
Creating new resource view if resource format is changed.

### CODE REVIEW

File: ckan/logic/action/update.py [hunk 1]
```diff
@@ -62,6 +62,8 @@ def resource_update(context, data_dict):
     model = context['model']
     user = context['user']
     id = _get_or_bust(data_dict, "id")
+    old_resource = _get_action('resource_show')(context, {'id': id})
+
```
Comment (smotornyuk) [inline ckan/logic/action/update.py]: I think that you should allow CKAN to perform all its default checks first and obtain old information no sooner than required

Comment (smotornyuk) [inline ckan/logic/action/update.py]: Or you can even use resource's model...

File: ckan/logic/action/update.py [hunk 2]
```diff
@@ -120,6 +122,12 @@ def resource_update(context, data_dict):
 
     resource = _get_action('resource_show')(context, {'id': id})
 
+    if old_resource['format'] != resource['format']:
+        _get_action('package_create_default_resource_views')(
+            {'model': context['model'], 'user': context['user'],
+             'ignore_auth': True},
+            {'package': updated_pkg_dict})
```
Comment (smotornyuk) [inline ckan/logic/action/update.py]: Wouldn't `resource_create_default_resource_view` fit here better?

**Comment by smotornyuk** [comment]
Now it looks ok :+1: 
@amercader , what do you think? It replicates logic from `resource_create` if format changed and tests are provided, so i have no objection against this PR

**Comment by amercader** [comment]
Looks great, thanks @Engerrs.

Feel free to merge @smotornyuk. Also marked for backporting.

**Pull request merged**